go/cmd/zk/internal/zkfs/zkfs_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package zkfs
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/z-division/go-zookeeper/zk"
+)
+
+func TestFS_CopyContext(t *testing.T) {
+	// fs := &FS{}
+
+	// Create a temporary directories for testing
+	srcPath, err := os.MkdirTemp("", "source")
+	assert.NoError(t, err)
+	defer os.RemoveAll(srcPath)
+
+	dstPath, err := os.MkdirTemp("", "destination")
+	assert.NoError(t, err)
+	defer os.RemoveAll(dstPath)
+
+}
+
+func TestIsFile(t *testing.T) {
+	assert.True(t, IsFile("/zk/somepath"))
+	assert.False(t, IsFile("/nonzk/somepath"))
+	assert.False(t, IsFile("nonzkpath"))
+}
+
+func TestParsePermMode(t *testing.T) {
+	assert.Equal(t, int32(0), ParsePermMode("zk"))
+	assert.Equal(t, int32(zk.PermRead|zk.PermWrite), ParsePermMode("zkrw"))
+	assert.Equal(t, int32(zk.PermRead|zk.PermWrite|zk.PermAdmin), ParsePermMode("zkrwa"))
+	assert.PanicsWithValue(t, "invalid mode", func() {
+		ParsePermMode("")
+		ParsePermMode("z")
+	})
```

Comment:
The two `ParsePermMode()` calls should be in separate assert calls. Currently code is testing that either one of them panics, but I think you want to test that both panic.

Reply:
Nice catch! Thanks!